homeassistant/components/mqtt/__init__.py

```diff
@@ -296,7 +296,8 @@ def async_setup(hass, config):
     if conf is None:
         conf = CONFIG_SCHEMA({DOMAIN: {}})[DOMAIN]
 
-    client_id = conf.get(CONF_CLIENT_ID)
+    # hbmqtt requires a client id to be set, so provide fallback
+    client_id = conf.get(CONF_CLIENT_ID, 'home-assistant')
```

Comment:
Please put 'home-assistant' in a constant and put it as a default value in the schema.